template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -961,4 +964,27 @@ public Optional<ApiClusterTemplateConfig> getRoleConfig(String serviceType, Stri
                         .filter(config -> configName.equals(config.getName()))
                         .findAny());
     }
+
+    public List<SafetyValve> getSafetyValves() {
+        List<SafetyValve> safetyValves = new ArrayList<>();
+        for (ApiClusterTemplateService apiClusterTemplateService : cmTemplate.getServices()) {
+            (!CollectionUtils.isEmpty(apiClusterTemplateService.getServiceConfigs()) ?
+                    apiClusterTemplateService.getServiceConfigs() : Collections.<ApiClusterTemplateConfig>emptyList()).stream()
```

Comment:
* Please extract the conditional expression into a local variable. The current formulation makes it harder to grasp what is the target of `stream()`. :smile: 
* Not sure the `<ApiClusterTemplateConfig>` is explicitly required, or could be simplified to the diamond operator `<>`.